src/app/_components/BlockList/LayoutA/Blocks.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { Icon } from '../../../../ui/Icon';
+import { Stack } from '../../../../ui/Stack';
+import { StxIcon } from '../../../../ui/icons';
+import { UIBlock, UIBlockType } from '../types';
+import { BlockCount } from './BlockCount';
+import { BurnBlock } from './BurnBlock';
+import { StxBlock } from './StxBlock';
+import { FADE_DURATION } from './consts';
+
+export function Blocks({ blockList, fadeEffect }: { blockList: UIBlock[]; fadeEffect: boolean }) {
+  return (
+    <Stack
+      pl={4}
+      pr={2}
+      gap={0}
+      width={'full'}
+      style={{
+        transition: `opacity ${FADE_DURATION / 1000}s`,
+        opacity: fadeEffect ? 0 : 1,
+      }}
+    >
+      {blockList.map((block, i) => {
+        switch (block.type) {
+          case UIBlockType.Block:
+            return (
+              <StxBlock
+                key={block.hash}
+                hash={block.hash}
+                height={block.height}
+                timestamp={block.timestamp}
+                txsCount={block.txsCount}
+                icon={
+                  i === 0 || (i > 0 && blockList[i - 1].type === UIBlockType.BurnBlock) ? (
```

Comment:
Would we be able to succintly define the condition in English. This is hard to decipher